R/pwr_2var.R

```diff
@@ -1,728 +1,1149 @@
-        d = d,
-        sig.level = alpha,
+  sign <- ifelse(d > 1, "\u2265", "\u2264")
+
+
+  if (calc == "sampleSize") {
+    str <- gettextf(
+      "We would need %1$s to reliably (with probability greater than %2$s) detect a variance ratio of %3$s, assuming a %4$s criterion for detection that allows for a maximum Type I error rate of %5$s.",
+      n_text, power, paste0("<i>\u03C1", sign, "</i>", d), tail_text, paste0("<i>\u03B1=</i>", alpha)
+    )
+  } else if (calc == "effectSize") {
+    str <- gettextf(
+      "A design with %1$s will reliably (with probability greater than %2$s) detect variance ratios of %3$s, assuming a %4$s criterion for detection that allows for a maximum Type I error rate of %5$s.",
+      n_text, power, paste0("<i>\u03C1", sign, "</i>", round(d, 3)), tail_text, paste0("<i>\u03B1=</i>", alpha)
+    )
+  } else if (calc == "power") {
+    str <- gettextf(
+      "A design with %1$s can detect variance ratios of %2$s with a probability of at least %3$s, assuming a %4$s criterion for detection that allows for a maximum Type I error rate of %5$s.",
+      n_text, paste0("<i>\u03C1", sign, "</i>", round(d, 3)), round(power, 3), tail_text, paste0("<i>\u03B1=</i>", alpha)
+    )
+  }
+
+  hypo_text <- ifelse(alt == "two.sided",
+    "<i>\u03C1</i>\u2260",
+    ifelse(alt == "less", "<i>\u03C1<1</i>", "<i>\u03C1>1</i>")
+  )
+
+  str <- paste0(
+    str,
+    "<p>",
+    gettextf(
+      "To evaluate the design specified in the table, we can consider how sensitive it is to true effects of increasing sizes; that is, are we likely to correctly conclude that %1$s when the variance ratio is extreme enough to care about?",
+      hypo_text
+    ),
+    "</p>"
+  )
+
+  html[["text"]] <- str
+
+
+  probs <- c(.5, .8, .95)
+  probs_es <- try(sapply(probs, function(p) {
+    if (alt == "two.sided" && d < 1) {
+      .pwr2Var2NTest(
+        n = n1, n.ratio = n_ratio, sig.level = alpha, power = p,
         alternative = alt
-      )
+      )$rho[2]
+    } else {
+      .pwr2Var2NTest(
+        n = n1, n.ratio = n_ratio, sig.level = alpha, power = p,
+        alternative = alt
+      )$rho[1]
+    }
+  }))
+  if (jaspBase::isTryError(probs_es)) {
+    table$setError(gettext("The specified design leads to (an) unsolvable equation(s) while computing the values for this power table. Try to enter less extreme values for the parameters."))
+    return()
+  }
+  sign1 <- ifelse(alt == "less" || (alt == "two.sided" && d < 1), ">", "<")
+  sign2 <- ifelse(alt == "less" || (alt == "two.sided" && d < 1), "\u2265", "\u2264")
+  sign3 <- ifelse(alt == "less" || (alt == "two.sided" && d < 1), "\u2264", "\u2265")
+
+  esText <- c(
+    gettextf("1 %1$s %2$s %3$s  %4$s", sign1, "\u03C1", sign2, format(round(as.numeric(probs_es[1]), 3), nsmall = 3)),
+    gettextf("%1$s %2$s %3$s %4$s %5$s", format(round(as.numeric(probs_es[1]), 3), nsmall = 3), sign1, "\u03C1", sign2, format(round(as.numeric(probs_es[2]), 3), nsmall = 3)),
```

Comment:
These seem also not need to be translated, because translator will do nothing for them. but it's ok if you don't have time to clean these.

Reply:
Good catch! I replaced gettextf with sprintf in these cases (couldn't be bothered to change it into paste, but maybe @LSLindeloo can update parts here if the code around it is being changed anyways at some point, AFAIK he did want to do some minor changes there)

Reply:
Also FYI if anyone runs into a similar problem in the future, I was able to easiliy find these strings by searching for the ones with a high ratio of `%` compared to overall text length 😄